core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -598,6 +598,15 @@ object SparkSubmit extends CommandLineUtils with Logging {
       }
     }
 
+    // In case of shells, spark.ui.showConsoleProgress can be true by default or by user.
+    if (isShell(args.primaryResource)) {
+      if (!sparkConf.contains("spark.ui.showConsoleProgress")) {
```

Comment:
Add a config constant now you're using this in more places?

Reply:
Thank you, @vanzin . I'll create a new config for this.

Reply:
Ur, @vanzin . It seems there is no pre-defined config for `spark.ui.XXX`.
Is it a convention for Apache Spark? Which class do you prefer to add a new config?

Reply:
It should be added to the config package object like all other constants in core.